airflow/providers/google/suite/hooks/drive.py

```diff
@@ -243,8 +300,21 @@ def upload_file(
             .create(body=file_metadata, media_body=media, fields="id", supportsAllDrives=True)
             .execute(num_retries=self.num_retries)
         )
-        self.log.info("File %s uploaded to gdrive://%s.", local_location, remote_location)
-        return file.get("id")
+        file_id = file.get("id")
+
+        upload_location = remote_location
+
+        if folder_id != "root":
+            try:
+                upload_location = self._resolve_file_path(folder_id)
+            except (GoogleApiClientError, AirflowException) as e:
```

Comment:
@eladkal Here is the catch for all exceptions - we don't want to fail the job due to path resolution

Reply:
Actually I think we shouldn't do that. The change you made to `_get_file_info` is enough.
AirflowException is not relevant here and as for GoogleApiClientError if this is raised after the 2 retries set then I guess we want the task to fail.. not to catch the exception and to just log it.

So lets revert this part... and I think we good to go.

Reply:
I disagree, because of the order of tasks. First file is uploaded and then we are trying to resolve its path, so if we fail on path resolution then file got uploaded, we failed the task due to unexpected behavior and we can't revert this change. So in my head there are 2 options:
1. We keep the solution as is
2. I move the path resolution code before uploading a file so the transaction doesn't need to be reverted.

Reply:
I agree with @aru-trackunit. I don't think the task should fail simply because the file path can't be resolved for logging purposes. It's unfortunate the Google API doesn't surface this information easily. As long as the task does what it's supposed to do, Airflow shouldn't care what is in its user-facing task logs. Having this logic configurable also helps too.

I do agree with @eladkal though that `AirflowException` is irrelevant. The operator is not failing for some orchestration or execution-related issue (IMO I think we use `AirflowException` with too broad of a brush sometimes). Maybe a custom exception is raised (e.g. `NestedFolderLimitReachedException` or something similar) just in case some real AirflowException is swallowed unknowingly?